django/contrib/admin/views/autocomplete.py

```diff
@@ -24,12 +24,15 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
+        self.id_field = request.GET.get('id_field', 'pk')
+        # set default id_field to 'pk' if not provided or not to_field_allowed()
+        #self.id_field = self.id_field if self.model_admin.to_field_allowed(request, self.id_field) else 'pk' TODO: not working
```

Comment:
Stick to the existing pattern to validate `to_field`

```python
to_field = request.GET.get(TO_FIELD_VAR)
if to_field:
    if not self.model_admin.to_field_allowed(request, to_field):
        raise DisallowedModelAdminToField("The field %s cannot be referenced." % to_field)
else:
    to_field = 'pk'
```

See https://github.com/django/django/blob/0cd465b63aa7c03a3d14bd5fd6543628d585f8da/django/contrib/admin/options.py#L1529-L1531

Reply:
I implemented things as you suggested however I can't get the unit test `test_autocomplete_view` to pass since it raises this error. 

I don't understand why the field `fk` on the model [`ParentWithFK`](https://github.com/django/django/blob/master/tests/admin_views/models.py#L891) is diallowed.